### PR TITLE
Copy of  "Dynamic Composable Kernel Initial Integration #1071"

### PR DESCRIPTION
**Pull request by asroy**
Create a new PR for the PR that has been reverted.

https://github.com/ROCmSoftwarePlatform/MIOpen/pull/1071

### CODE REVIEW

**Review by junliume — APPROVED** [review]

File: src/composable_kernel/host/host_tensor/include/device.hpp [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+    hipLaunchKernelGGL(kernel, grid_dim, block_dim, lds_byte, stream_id, args...);
+}
+
+template <typename... Args, typename F>
+float launch_and_time_kernel(
+    F kernel, int nrepeat, dim3 grid_dim, dim3 block_dim, std::size_t lds_byte, Args... args)
+{
+    KernelTimer timer;
+
+    printf("%s: grid_dim {%d, %d, %d}, block_dim {%d, %d, %d} \n",
```
Comment (atamazov) [inline src/composable_kernel/host/host_tensor/include/device.hpp:50]: ⚓ This should be controlled by MIOPEN_LOG_LEVEL etc.

File: src/composable_kernel/composable_kernel/include/tensor_operation/blockwise_gemm_xdlops.hpp [hunk 2]
```diff
@@ -0,0 +1,524 @@
...
+{
+
+    using CIndex = MultiIndex<2>;
+
+    static constexpr auto I0 = Number<0>{};
+    static constexpr auto I1 = Number<1>{};
+    static constexpr auto I2 = Number<2>{};
+    static constexpr auto I3 = Number<3>{};
+
+    static constexpr index_t WaveSize = 64;
```
Comment (atamazov) [inline src/composable_kernel/composable_kernel/include/tensor_operation/blockwise_gemm_xdlops.hpp:27]: 64 is invalid for gfx10+

File: src/composable_kernel/composable_kernel/include/tensor_operation/blockwise_gemm_xdlops.hpp [hunk 3]
```diff
@@ -0,0 +1,524 @@
...
+    using CIndex = MultiIndex<2>;
+
+    static constexpr auto I0 = Number<0>{};
+    static constexpr auto I1 = Number<1>{};
+    static constexpr auto I2 = Number<2>{};
+    static constexpr auto I3 = Number<3>{};
+
+    static constexpr auto xdlops_gemm = XdlopsGemm<float, MPerWave, NPerWave, K1>{};
+
+    static constexpr index_t WaveSize = 64;
```
Comment (atamazov) [inline src/composable_kernel/composable_kernel/include/tensor_operation/blockwise_gemm_xdlops.hpp:247]: Same

File: src/composable_kernel/composable_kernel/include/tensor_operation/blockwise_gemm_xdlops.hpp (hunk 2 above)
Comment (junliume) [reply src/composable_kernel/composable_kernel/include/tensor_operation/blockwise_gemm_xdlops.hpp:27]: @asroy @zjing14 @carlushuang could you take a look at this?

**Pull request merged**